server/schedulers/balance_leader.go

```diff
@@ -174,13 +160,20 @@ func (l *balanceLeaderScheduler) transferLeaderOut(cluster opt.Cluster, source *
 		schedulerCounter.WithLabelValues(l.GetName(), "no-leader-region").Inc()
 		return nil
 	}
-	target := l.selector.SelectTarget(cluster, cluster.GetFollowerStores(region))
-	if target == nil {
-		log.Debug("region has no target store", zap.String("scheduler", l.GetName()), zap.Uint64("region-id", region.GetID()))
-		schedulerCounter.WithLabelValues(l.GetName(), "no-target-store").Inc()
-		return nil
+	targets := cluster.GetFollowerStores(region)
+	targets = filter.SelectTargetStores(targets, l.filters, cluster)
+	leaderScheduleStrategy := l.opController.GetLeaderScheduleStrategy()
+	sort.Slice(targets, func(i, j int) bool {
+		return targets[i].LeaderScore(leaderScheduleStrategy, 0) < targets[j].LeaderScore(leaderScheduleStrategy, 0)
+	})
+	for _, target := range targets {
+		if op := l.createOperator(cluster, region, source, target); len(op) > 0 {
+			return op
+		}
 	}
-	return l.createOperator(cluster, region, source, target)
+	log.Debug("region has no target store", zap.String("scheduler", l.GetName()), zap.Uint64("region-id", region.GetID()))
+	schedulerCounter.WithLabelValues(l.GetName(), "no-target-store").Inc()
+	return nil
 }
 
 // transferLeaderIn transfers leader to the target store.
```

Comment:
Do we need to update `transferLeaderIn`?

Reply:
we don't because after selecting the region, there is only 1 peer may be selected as source (the region leader).